PrimeRust/solution_1/src/main.rs

```diff
@@ -283,16 +283,12 @@ pub mod primes {
 
         #[inline(always)]
         fn is_num_flagged(&self, number: usize) -> bool {
-            if number % 2 == 0 {
-                return false;
-            }
-            let index = number / 2;
-            self.flags.get(index)
+            self.flags.get(number)
```

Comment:
I think we can get rid of the `is_num_flagged` function entirely now :)

It was originally meant to be a function that returned true/false based on a supplied whole number. With your refactoring, it doesn't do that any more -- we're supplying a storage index instead.

Reply:
Actually -- just noted that this is used in `print_results_stderr`, and this change would stop that from working, since the meaning of the input has changed.

The best approach would probably be: 
1. to leave `is_num_flagged` as it was
2. in your changes, call `self.flags.get(*v)` instead of calling `is_num_flagged`

That way we get to preserve the behaviour with the printout etc, but get to use your cleaner code for the search and counting.

Reply:
You are right. I was under the impression that `print_results_stderr` still needed it, but I rechecked after your comment and the function indeed has access to the private fields of `PrimeSieve` because it is in the same module.